packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,855 @@
+///
+/// An example usage might look like:
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   preferredSize: Size(800, 600),
+///   preferredConstraints: BoxConstraints(minWidth: 640, minHeight: 480),
+///   title: "Example Window",
+/// );
+/// runWidget(
+///   RegularWindow(
+///     controller: controller,
+///     child: MaterialApp(home: Container()),
+///   ),
+/// );
+/// ```
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+class RegularWindow extends StatelessWidget {
+  /// Creates a regular window widget.
+  ///
+  /// The [controller] creates the native backing window into which the
+  /// [child] widget is rendered.
+  ///
+  /// It is up to the caller to destroy the window by calling
+  /// [RegularWindowController.destroy] when the window is no longer needed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  RegularWindow({super.key, required this.controller, required this.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+
+  /// Controller for this widget.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final RegularWindowController controller;
+
+  /// The content rendered into this window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final Widget child;
+
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  @override
+  Widget build(BuildContext context) {
+    return View(
+      view: controller.rootView,
+      child: WindowControllerScope(controller: controller, child: child),
+    );
+  }
+}
+
+enum _WindowControllerAspect { contentSize, title, activated, maximized, minimized, fullscreen }
+
+/// Provides descendants with access to the [BaseWindowController] associated with
+/// the window that is being rendered.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+///  * [RegularWindow], the widget to create a regular window.
+@internal
+class WindowControllerScope extends InheritedModel<_WindowControllerAspect> {
+  /// Creates a new [WindowControllerScope].
+  ///
+  /// This widget is used by the window widgets to provide
+  /// widgets in a window's subtree with access to information about
+  /// the window.
+  ///
+  /// The [controller] is the controller associated with this window,
+  /// and the [child] is the widget tree that will have access
+  /// to this context.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  WindowControllerScope({super.key, required this.controller, required super.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+
+  /// The controller associated with this window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final BaseWindowController controller;
+
+  /// Returns the [WindowControllerScope].
```

Comment:
This should be `BaseWindowController`. I'd consider adding a quick explanation as to what you can do with the controller.

```suggestion
  /// Returns the [BaseWindowController] for the window that hosts the given context.
```